File: frontend/src/models/Product/Product.js
export default class Product{
    constructor(jsonObj){
        if (jsonObj){
            this.id = jsonObj.id;
            this.name = jsonObj.name;
            this.description = jsonObj.description;
            this.price = jsonObj.price;
            this.quantity = jsonObj.quantity;
        }
    }
}

File: backend/Product/src/main/java/tecnico/ulisboa/pt/Product/domain/Product.java
package tecnico.ulisboa.pt.Product.domain;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;


@Entity
public class Product {
    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE)
    @Column(name = "id", nullable = false)
    @Getter
    @Setter
    private Integer id;

    @Getter
    @Setter
    private String name;

    @Getter
    @Setter
    private Integer price;

    @Getter
    @Setter
    private Integer quantity;

    public Product() {}

    public Product(String name, Integer price, Integer quantity) {
        this.name = name;
        this.price = price;
        this.quantity = quantity;
    }


}